s3/src/test/scala/akka/stream/alpakka/s3/scaladsl/S3IntegrationSpec.scala

```diff
@@ -374,6 +375,46 @@ trait S3IntegrationSpec
     }
   }
 
+  @tailrec
+  final def fillUntilMinChunkSize(curr: String): String = {
+    if (curr.getBytes("UTF-8").length < S3.MinChunkSize)
+      fillUntilMinChunkSize(s"$curr\n$curr")
+    else
+      curr
+  }
+
+  it should "upload 1 file slowly, cancel it and retrieve a multipart upload" in {
```

Comment:
So in general this test is designed to cancel the multipart upload roughly halfway through and then see that both the `MultipartUploadResults` and the `ListParts` give sane results.

Since we need to cancel halfway through, we manually construct a throttled source and we have to make sure that each element in the source is at least as big as the min chunk size in order for the state to persist in  S3.